Userland/Libraries/LibGfx/Rect.h

```diff
@@ -479,6 +479,15 @@ class Rect {
         return *this;
     }
 
+    [[nodiscard]] Rect<T> operator/(T factor) const { return { m_location / factor, m_size / factor }; }
+
+    Rect<T>& operator/=(T factor)
+    {
+        m_location /= factor;
+        m_size /= factor;
+        return *this;
+    }
```

Comment:
I'm pretty sure it's the kind of operator, for which I need to look at its definition to know what it does for the location.

And it seems that `scale_by` already does the same thing (same for Size). I understand that you can't use that for your use case, as you can't multiply by 0.5 (or any 1 / (1 << n)), but this is IMO a hint that it shouldn't be an operator. So I guess that we might want `scale_[up,down]_by` variants for int-based `Rect`, or you could either use the `set_size` method like this:
```cpp
auto ll_rect = [=](int r, int) {
    return tile_rect.set_size(tile_rect.size() / (1 << (max_number_of_decomposition_levels - r)));
};
```

Reply:
I want to scale down the location too though.

`operator/=` does the same thing as `operator*=`. Isn't having both self-consistent? Do you have a problem with `*=` too?

Reply:
I didn't realize that we already had `operator*=`, and yes I don't like it either. But I guess this was already discussed at the time, so I should probably learn to live with the knowledge of its existence instead.

> I want to scale down the location too though.

Where are you doing that :thinking:, I wasn't able to find where this was used outside of tests, and there it seems that the location of `tile_rect` is initialized and stays to null. (Not important)

Reply:
Yes,. it's only used in tests. I suppose the location is 0, 0 there and it's not important that it's getting scaled, but it's supposed to be consistent with `reference_grid_coordinates_for_ll_band()` in JPEG2000Loader.cpp, and that does scale both origin and size (using formulas from the spec).

Reply:
Oh ok, that's why. Thanks for clarifying that!